sdk/storage/storage-file-datalake/src/clients.ts

```diff
@@ -1111,12 +1111,18 @@ export class DataLakeFileClient extends DataLakePathClient {
   ): Promise<FileUploadResponse> {
     const { span, spanOptions } = createSpan("DataLakeFileClient-upload", options.tracingOptions);
     try {
-      if (isNode && data instanceof Buffer) {
+      if (isNode) {
+        let length = 0;
+        if (data instanceof ArrayBuffer) {
+          length = data.byteLength;
+        } else {
+          length = (data as any).length;
+        }
         return this.uploadData(
           (offset: number, size: number): Buffer => {
-            return data.slice(offset, offset + size);
+            return (data as any).slice(offset, offset + size);
```

Comment:
Why use any here? We should avoid using any type which will loose benifit of Typing checks.